### PR TITLE
Feat/leaderboard

### PR DESCRIPTION
**Pull request by alvaronietos**

### CODE REVIEW

File: src/scenes/LeaderboardScene.js [hunk 1]
```diff
@@ -79,6 +79,9 @@ export class LeaderboardScene extends Phaser.Scene {
     this.statusText.destroy();
     this.statusText = null;
 
+    const user = this.game.registry.get('user');
+    const currentNickname = user?.user_metadata?.nickname;
```
Comment (simon0191) [inline src/scenes/LeaderboardScene.js]: **Suggestion**: Matching by nickname works today because `nickname` has a `UNIQUE` constraint, but it's still an indirect comparison — if `user_metadata.nickname` ever diverges from `profiles.nickname` (e.g., a profile sync failure on first login, or a future nickname-update feature), the highlight would silently break.

A more robust approach would be to match by user ID:

```js
const userId = user?.id;
```

```js
const isCurrentUser = userId && row.id === userId;
```

This would require adding `id` to the leaderboard API SELECT:

```sql
SELECT
  id,
  COALESCE(nickname, 'Anónimo') AS nickname,
  ...
```

Minor nit — not blocking, since for this friend group nicknames are stable and unique. But worth considering if the leaderboard grows.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/scenes/LeaderboardScene.js [hunk 2]
```diff
@@ -97,11 +100,13 @@ export class LeaderboardScene extends Phaser.Scene {
       const winRate = `${row.win_rate}%`.padStart(8, ' ');
       const line = `${rank}${name}${wins}${losses}${winRate}`;
 
+      const isCurrentUser = currentNickname && row.nickname === currentNickname;
```
Comment (simon0191) [inline src/scenes/LeaderboardScene.js]: Good use of short-circuit — correctly handles guest mode (`user` is null) and missing metadata (no `nickname` key). No row gets highlighted in those cases, which is the right behavior.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by simon0191 — COMMENTED** [review]
## Review Summary

Nice, small, focused change — highlights the current user's row in the leaderboard with yellow (`#ffcc00`). Clean implementation that follows existing patterns.

### What's good
- **Follows existing conventions**: Uses `user?.user_metadata?.nickname` — same pattern as `TitleScene.js:73`.
- **Guest mode safe**: `currentNickname` is falsy when no user is logged in, so the short-circuit in `isCurrentUser` correctly skips highlighting.
- **Performance**: User lookup happens once before the loop, not per-row.
- **CI is green**: All checks pass (test, e2e, Vercel preview).

### Observations
- **Nickname matching is fragile long-term**: The comparison relies on `user_metadata.nickname` matching `profiles.nickname`. Today this works because: (1) nicknames are `UNIQUE` in the DB, and (2) `syncProfile` sets the profile nickname from `user_metadata` on first creation. However, if the profile sync fails on first login (caught and swallowed at `LoginScene.js:56-58`), or if a nickname-update feature is ever added, these could diverge silently. Matching by user `id` via the leaderboard API would be more robust. See inline comment.
- **No test coverage gap**: The change is purely presentational in a Phaser scene. Consistent with the project's approach of only testing pure logic modules.
- **No new edge case bugs found**: `COALESCE(nickname, 'Anónimo')` in the SQL means null-nickname profiles show as 'Anónimo', and `currentNickname` would be falsy in that case — so no false match.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/scenes/LeaderboardScene.js (hunk 1 above)
Comment (simon0191) [inline src/scenes/LeaderboardScene.js]: Please match by user ID (`user?.id`) instead of nickname. The leaderboard API should include the profile `id` in its response, then compare `row.id === userId` here.

Nickname matching is fragile — if profile sync fails on first login (error is swallowed at `LoginScene.js:56-58`) or a nickname-update feature is added later, `user_metadata.nickname` and `profiles.nickname` could diverge silently.

Suggested changes:

**`api/leaderboard.js`** — add `id` to the SELECT:
```sql
SELECT
  id,
  COALESCE(nickname, 'Anónimo') AS nickname,
  ...
```

**`LeaderboardScene.js`**:
```js
const userId = user?.id;
// ...
const isCurrentUser = userId && row.id === userId;
```

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by simon0191 — CHANGES_REQUESTED** [review]
## Review Summary

Nice, small, focused change — highlights the current user's leaderboard row in yellow (`#ffcc00`). Clean implementation that follows existing patterns.

### What's good
- **Follows existing conventions**: Uses `user?.user_metadata?.nickname` — same pattern as `TitleScene.js:73`.
- **Guest mode safe**: `currentNickname` is falsy when no user is logged in, so the short-circuit in `isCurrentUser` correctly skips highlighting.
- **Performance**: User lookup happens once before the loop, not per-row.
- **CI is green**: All checks pass (test, e2e, Vercel preview).

### Requested change

**Match by user ID instead of nickname.** The current approach compares `user_metadata.nickname` against `profiles.nickname`. While nicknames are `UNIQUE` in the DB, this is still an indirect comparison that can silently break if:
1. Profile sync fails on first login (the error is caught and swallowed at `LoginScene.js:56-58`)
2. A nickname-update feature is ever added

The fix is straightforward:
- **`api/leaderboard.js`**: Add `id` to the SELECT (`SELECT id, COALESCE(nickname, 'Anónimo') AS nickname, ...`)
- **`LeaderboardScene.js`**: Compare by ID instead of nickname (`user?.id` vs `row.id`)

This is robust regardless of how nicknames evolve. See inline comment for details.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by simon0191 — APPROVED** [review]
